app/models/king.rb

```diff
@@ -1,2 +1,32 @@
 class King < Piece
+  # valid moves--
+  # king may move one square any direction
+  # must NOT move onto square that is being attacked by opposing pieces
+  # may capture piece IF where it's moving to there is no other piece protecting soon-to-be-captured piece(in-check)
+  # because king may NOT move onto square that is being attacked 2 kings may NEVER stand next to each other => position: illegal
+
+  def valid_move?(x, y)
+    return false unless position_on_board?(x, y)
+    return false if move_too_far?(x, y)
+    return false if move_into_position_under_attack?(x, y)
+    true
+  end
+
+  def position_on_board?(x, y)
```

Comment:
One thing you can do here is make `position_on_board`, `move_too_far` and `move_into_position_under_attack` private methods.

Rule I use: If the method is never used outside of the Object. Then I make it private. :smile:

More info: http://raganwald.com/2014/02/11/private-methods-in-ruby.html